.github/workflows/ci.yml

```diff
@@ -83,7 +83,7 @@ jobs:
   compatibility-test:
     strategy:
       matrix:
-        go-version: ["~1.21.1", "~1.20.8"]
```

Comment:
Why no leave the minimums to ensure security issues are not re-introduced? What has changed since [this](https://github.com/open-telemetry/opentelemetry-go/pull/4480#issuecomment-1710333713)?

Reply:
I do not think it is that much relevant as we are not producing binaries.

I created https://github.com/open-telemetry/opentelemetry-go/issues/4639 to have the discussion in one place.

Reply:
Addressed https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4418/commits/187c8159ee295dbfd6e883d1199e85712c4fa926